File: src/main/java/com/checkers/exceptions/CellIsBusyException.java
package com.checkers.exceptions;

import com.checkers.board.Cell;

public class CellIsBusyException extends RuntimeException {

    private static final String MSG_TEMPLATE = "Cell %s already has the piece";

    public CellIsBusyException(Cell cell) {
        super(MSG_TEMPLATE.formatted(cell));
    }
}

File: src/main/kotlin/com/checkers/exception/CellIsBusyException.kt
package com.checkers.exception

import com.checkers.board.Cell
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus

@ResponseStatus(HttpStatus.BAD_REQUEST)
class CellIsBusyException(cell: Cell) : RuntimeException("Cell $cell already has the piece")
